svj_ntuple_processing/__init__.py

```diff
@@ -588,170 +577,163 @@ def veto_HEM(eta,phi,pt):
                 ) < .2]
-        cutflow['nmuons=1'] = len(a)
-        a = a[a['NElectrons']==0]
-        cutflow['nelectrons=0'] = len(a)
-    else:
-        # lepton vetoes
-        a = a[a['NMuons']==0]
-        cutflow['nmuons=0'] = len(a)
-        a = a[a['NElectrons']==0]
-        cutflow['nelectrons=0'] = len(a)
+    copy.cutflow['nmuons=1'] = len(a)
+    a = a[a['NElectrons']==0]
+    copy.cutflow['nelectrons=0'] = len(a)
+    copy.array = a
+    return copy
+
+
+def select_muon_veto(arrays:Arrays):
+    # lepton vetoes
+    arrays.array = arrays.array[arrays.array['NMuons']==0]
+    arrays.cutflow['nmuons=0'] = len(arrays.array)
+    return arrays
+
+def select_electron_veto(arrays:Arrays):
+    arrays.array = arrays.array[arrays.array['NElectrons']==0]
+    arrays.cutflow['nelectrons=0'] = len(arrays.array)
+    return arrays
+
 
+def select_metdphi(arrays:Arrays):
+    a = arrays.array
+    METDphi = calc_dphi(a['JetsAK15.fCoordinates.fPhi'][:,1].to_numpy(), a['METPhi'].to_numpy())
+    a = a[abs(METDphi)<1.5]
+    arrays.cutflow['abs(metdphi)<1.5'] = len(a)
+    arrays.array = a
+    return arrays
+
+def select_metfilter_standard(arrays:Arrays):
     # MET filters
-    for b in met_filters:
-        a = a[a[b]!=0] # Pass events if not 0, is that correct?
-    cutflow['metfilter'] = len(a)
+    # following the order from https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETOptionalFiltersRun2#Run_2_recommendations
+    met_filters = [
+        'PrimaryVertexFilter',
+        'globalSuperTightHalo2016Filter',
+        'HBHENoiseFilter',
+        'HBHEIsoNoiseFilter',
+        'BadPFMuonFilter',
+        'BadPFMuonDzFilter',
+        'eeBadScFilter',
+        'ecalBadCalibFilter',
+    ]
+    for f in met_filters:
+        arrays.array = arrays.array[arrays.array[b]!=0] # Pass events if not 0, is that correct?
+    arrays.cutflow['metfilter'] = len(arrays.array)
+    return arrays
 
+
+def select_highpt_muon_veto(arrays:Arrays):
+    # muon pt < 1500 filter to avoid highMET events
+    arrays.array = arrays.array[~ak.any(arrays.array['Muons.fCoordinates.fPt'] > 1500., axis=-1)]
+    arrays.cutflow['muonpt<1500'] = len(arrays.array)
+    return arrays
+
+
+def select_deadcell_prerequisite(arrays: Arrays):
+    # TODO: This selection was inconsistent between various items
+    # Below was the version found in selection_deadcells
+    arrays.array = arrays.array[ak.count(arrays.array['Jets.fCoordinates.fPt'], axis=-1) >= 2]
+    arrays.cutflow['n_ak4jets>=2'] = len(arrays.array)
+    return arrays
+
+    # ALternate function found in cr_filter_preselection
+    a = a[np.abs(a['Jets.fCoordinates.fEta'][:,0])<2.4]
+    a = a[np.abs(a['Jets.fCoordinates.fEta'][:,1])<2.4]
+    cutflow['ak4j1j2_eta<2.4'] = len(a)
+
+
+def select_metfilter_custom(arrays:Arrays):
+    a = arrays.array
     # test hf filter separately
     a = a[a['hfNoisyHitsFilter']!=0]
-    cutflow['hffilter'] = len(a)
+    arrays.cutflow['hffilter'] = len(a)
```

Comment:
this should go with the regular met filters, not the custom ones. (we added it late, so I kept it separate for testing. but the impact was as expected, so it can just go into the regular list of met filters now)